src/pages/RegisterPage.tsx

```diff
@@ -1,27 +1,89 @@
+import { zodResolver } from '@hookform/resolvers/zod'
 import { css } from '@styled-stytem/css'
+import { useCallback, useRef, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+import { z } from 'zod'
 
+import { useRegister } from '@/api/hooks/register'
+import EmailForm from '@/components/register/EmailForm'
+import PasswordForm from '@/components/register/PasswordForm'
+import StudentIdForm from '@/components/register/StudentIdForm'
+import UsernameForm from '@/components/register/UsernameForm'
+import Button from '@/components/ui/button'
+import { Form } from '@/components/ui/form'
 import { Input } from '@/components/ui/input'
 import { Label } from '@/components/ui/label'
+import { RegisterFormSchema } from '@/lib/zod/register-schema'
+import { ValidState } from '@/types/register'
 
 const RegisterPage = () => {
+  const file = useRef<HTMLInputElement>(null)
+  const [valid, setValid] = useState<ValidState>({ email: 'unknown', username: 'unknown', studentId: 'unknown' })
+
+  const { mutate: mutateRegister } = useRegister()
+  const navigate = useNavigate()
+  const form = useForm<z.infer<typeof RegisterFormSchema>>({
+    resolver: zodResolver(RegisterFormSchema),
+    defaultValues: {
+      email: '',
+      password: { password: '', confirm: '' },
+      username: '',
+      studentId: '',
+    },
+  })
+
+  const onSubmit = (values: z.infer<typeof RegisterFormSchema>) => {
+    if (!file.current?.files?.length) return
+    if (valid.email !== 'valid' || valid.username !== 'valid' || valid.studentId !== 'valid') return
+    mutateRegister(
+      {
+        screenShot: file.current.files[0],
+        email: values.email,
+        username: values.username,
+        password: values.password.password,
+        studentNumber: values.studentId,
+      },
+      {
+        onSuccess: () => {
+          alert('회원가입이 완료되었습니다.')
+          navigate('/login')
+        },
+      },
+    )
+  }
+
+  const handleValidation = useCallback((target: string, value: string) => {
+    setValid(v => ({ ...v, [target]: value }))
+  }, [])
+
```

Comment:
```suggestion
useEffect(() => {
    if (isAuth) navigate('/mypage')
  }, [isAuth, navigate])
```

LoginPage처럼 이미 로그인 되어 있을 경우의 예외처리

Reply:
아 이거 예외 처리 안 해둔 게 회원 가입 페이지에 대한 접근 루트가 불명확해서, 불필요한 경우가 있으려나? 싶어서 아직 그대로 둔 상태입니당.

Reply:
아 오케이
loginPage, registerPage, logout 기능 < 얘네들은 더 구체적인 기능명세서랑 유저플로우 나온담에 개선하면 될듯요